src/Altinn.Broker.Integrations/Altinn/Events/AltinnEventBus.cs

```diff
@@ -42,7 +42,7 @@ public AltinnEventBus(
 
     public async Task Publish(AltinnEventType type, string resourceId, string fileTransferId, string? subjectOrganizationNumber = null, CancellationToken cancellationToken = default)
     {
-        _backgroundJobClient.Enqueue(() => Publish(type, resourceId, fileTransferId, Guid.NewGuid(), DateTime.UtcNow, subjectOrganizationNumber, cancellationToken));
```

Comment:
The old implementation generated the Guid for the event entity when it was queued up on our end. By doing this, we implemented the outbox pattern to ensure we did not produce duplicate events. With the new version, a new Guid is created on every attempt and thus we may get duplicate events in cases where the event publish succeeds but we do not see it.

However, in the middleware we set "AutomaticRetry" to zero so we'll avoid that issue. But we also won't do any retries. If this means we end up with an exception from our Slackbot in #test-varslinger it is probably good enough for now, but we should look into other solutions in the longer term to make sure we also do retries.

Alternatively, you could change the implementation such that EventId is passed into EventBusMiddleware.Publish (and thus created in the _backgroundJobClient.Enqueue call) and remove AutomaticRetry.

Reply:
Will change ist to keep functionality as is.